addons/luamodel/luatypes.h

```diff
@@ -0,0 +1,385 @@
+                 vector_table[5].get<RigidBodyDynamics::Math::SpatialVector>()
+               );
+      break;
+    case 6:
+      result = RigidBodyDynamics::Joint(
+                 vector_table[1].get<RigidBodyDynamics::Math::SpatialVector>(),
+                 vector_table[2].get<RigidBodyDynamics::Math::SpatialVector>(),
+                 vector_table[3].get<RigidBodyDynamics::Math::SpatialVector>(),
+                 vector_table[4].get<RigidBodyDynamics::Math::SpatialVector>(),
+                 vector_table[5].get<RigidBodyDynamics::Math::SpatialVector>(),
+                 vector_table[6].get<RigidBodyDynamics::Math::SpatialVector>()
+               );
+      break;
+    //default:
+    //  throw RigidBodyDynamics::Errors::RBDLFileParseError(
+    //        "Invalid number of DOFs for joint.");
+    //  break;
+    }
+  }
+
+  stackRestore();
+
+  return result;
+}
+
+
+//==============================================================================
+template<>
+RigidBodyDynamics::Body
+LuaTableNode::getDefault<RigidBodyDynamics::Body>(
+    const RigidBodyDynamics::Body &default_value)
+{
+  RigidBodyDynamics::Body result = default_value;
+
+  if (stackQueryValue()) {
+    LuaTable vector_table = LuaTable::fromLuaState (luaTable->L);
+
+    double mass = 0.;
+    RigidBodyDynamics::Math::Vector3d com(
+          RigidBodyDynamics::Math::Vector3d::Zero(3));
+    RigidBodyDynamics::Math::Matrix3d inertia(
+          RigidBodyDynamics::Math::Matrix3d::Identity(3,3));
+
+    mass = vector_table["mass"];
+    com = vector_table["com"]
+        .getDefault<RigidBodyDynamics::Math::Vector3d>(com);
+    inertia = vector_table["inertia"]
+        .getDefault<RigidBodyDynamics::Math::Matrix3d>(inertia);
+
+    result = RigidBodyDynamics::Body (mass, com, inertia);
+  }
+
+  stackRestore();
+
+  return result;
+}
+
+//==============================================================================
+
+template<> Point
+LuaTableNode::getDefault<Point>(
+  const Point &default_value)
+{ 
+  Point result = default_value;
+
+  if (stackQueryValue()) {
+    LuaTable point_table = LuaTable::fromLuaState (luaTable->L);
+
+    result.name         = point_table["name"].get<std::string>();
+    result.point_local  = point_table["point"];
+    result.body_name    = point_table["body"].get<std::string>();
+  }
+
+  stackRestore();
+
+  return result;
+}
+
+//==============================================================================
+
+template<> MotionCaptureMarker
+LuaTableNode::getDefault<MotionCaptureMarker>(
+  const MotionCaptureMarker &default_value)
+{
+  MotionCaptureMarker result = default_value;
+
+  if (stackQueryValue()) {
+    LuaTable marker_table = LuaTable::fromLuaState (luaTable->L);
+
+    result.name         = marker_table["name"].get<std::string>();
+    result.point_local  = marker_table["point"]
+                            .getDefault<RigidBodyDynamics::Math::Vector3d>(
+                              result.point_local);
```

Comment:
Seems to be missing the `body` parameter.

Reply:
Yikes! Included now.